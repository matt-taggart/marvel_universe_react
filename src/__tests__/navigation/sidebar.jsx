import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SideBar from '../../components/navigation/sidebar';

Enzyme.configure({ adapter: new Adapter() })

describe('Sidebar Component', () => {
  it('Should render correctly', () => {
    const wrapper = shallow(<SideBar />);
    expect(wrapper).toMatchSnapshot();
  });
});
