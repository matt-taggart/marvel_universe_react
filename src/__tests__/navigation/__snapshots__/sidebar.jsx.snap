// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Component Should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="menu-label"
        >
          Browse
        </p>,
        <ul
          className="menu-list"
        >
          <li>
            <Link
              replace={false}
              to="/"
            >
              Characters
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/comics"
            >
              Comics
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/creators"
            >
              Creators
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/events"
            >
              Events
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/series"
            >
              Series
            </Link>
          </li>
        </ul>,
      ],
      "className": "menu column is-one-quarter",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Browse",
          "className": "menu-label",
        },
        "ref": null,
        "rendered": "Browse",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <Link
                replace={false}
                to="/"
              >
                Characters
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/comics"
              >
                Comics
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/creators"
              >
                Creators
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/events"
              >
                Events
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/series"
              >
                Series
              </Link>
            </li>,
          ],
          "className": "menu-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/"
              >
                Characters
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Characters",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Characters",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/comics"
              >
                Comics
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Comics",
                "replace": false,
                "to": "/comics",
              },
              "ref": null,
              "rendered": "Comics",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/creators"
              >
                Creators
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Creators",
                "replace": false,
                "to": "/creators",
              },
              "ref": null,
              "rendered": "Creators",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/events"
              >
                Events
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Events",
                "replace": false,
                "to": "/events",
              },
              "ref": null,
              "rendered": "Events",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/series"
              >
                Series
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Series",
                "replace": false,
                "to": "/series",
              },
              "ref": null,
              "rendered": "Series",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="menu-label"
          >
            Browse
          </p>,
          <ul
            className="menu-list"
          >
            <li>
              <Link
                replace={false}
                to="/"
              >
                Characters
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/comics"
              >
                Comics
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/creators"
              >
                Creators
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/events"
              >
                Events
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/series"
              >
                Series
              </Link>
            </li>
          </ul>,
        ],
        "className": "menu column is-one-quarter",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Browse",
            "className": "menu-label",
          },
          "ref": null,
          "rendered": "Browse",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <Link
                  replace={false}
                  to="/"
                >
                  Characters
                </Link>
              </li>,
              <li>
                <Link
                  replace={false}
                  to="/comics"
                >
                  Comics
                </Link>
              </li>,
              <li>
                <Link
                  replace={false}
                  to="/creators"
                >
                  Creators
                </Link>
              </li>,
              <li>
                <Link
                  replace={false}
                  to="/events"
                >
                  Events
                </Link>
              </li>,
              <li>
                <Link
                  replace={false}
                  to="/series"
                >
                  Series
                </Link>
              </li>,
            ],
            "className": "menu-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/"
                >
                  Characters
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Characters",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Characters",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/comics"
                >
                  Comics
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Comics",
                  "replace": false,
                  "to": "/comics",
                },
                "ref": null,
                "rendered": "Comics",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/creators"
                >
                  Creators
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Creators",
                  "replace": false,
                  "to": "/creators",
                },
                "ref": null,
                "rendered": "Creators",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/events"
                >
                  Events
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Events",
                  "replace": false,
                  "to": "/events",
                },
                "ref": null,
                "rendered": "Events",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/series"
                >
                  Series
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Series",
                  "replace": false,
                  "to": "/series",
                },
                "ref": null,
                "rendered": "Series",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
